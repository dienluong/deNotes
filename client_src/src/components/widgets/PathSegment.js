import React from 'react';

class PathSegment extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.props.onClick(this.props.index, this.props.label);
  }

  render() {
    return (
      <li className={ this.props.className } onClick={ this.handleClick }><span>{ this.props.label } &gt;</span></li>
    );
  }
}

export default PathSegment;
