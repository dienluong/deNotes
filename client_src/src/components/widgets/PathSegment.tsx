import React from 'react';

// Types
type PropsT = {
  className: string;
  label: string;
  onClick: (params: any) => any;
}

function PathSegment({ className, label, onClick: clickCallback }: PropsT) {
  return (
    <li className={ className } onClick={ clickCallback }><span>{ label }&gt;</span></li>
  );
}

export default PathSegment;
