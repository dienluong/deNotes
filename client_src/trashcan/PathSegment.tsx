import React from 'react';
import unescape from 'lodash/unescape';

// Types
type PropsT = {
  className: string;
  label: string;
  onClick: (params: any) => any;
}

function PathSegment({ className, label, onClick: clickCallback }: PropsT) {
  return (
    <li className={ className } onClick={ clickCallback }><span>{ unescape(label) }&gt;</span></li>
  );
}

export default PathSegment;
