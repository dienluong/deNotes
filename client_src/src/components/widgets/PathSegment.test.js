import PathSegment from './PathSegment';
import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';

afterEach(cleanup);

it('renders an element with a label and reacting to a click event', () => {
  const props = {
    className: 'test-css-class',
    label: 'test-segment-label',
    onClick: jest.fn(),
  };

  const { queryAllByText, container } = render(<PathSegment { ...props } />);
  expect(container).toMatchSnapshot();
  const elements = queryAllByText(props.label, { exact: false });
  expect(elements).toHaveLength(1);
  expect(container.firstChild).toHaveClass(props.className);
  fireEvent.click(container.firstChild);
  expect(props.onClick).toBeCalledTimes(1);
});
