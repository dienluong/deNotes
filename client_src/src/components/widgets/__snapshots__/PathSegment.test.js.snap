// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an element with a label and reacting to a click event 1`] = `
<div>
  <li
    class="test-css-class"
  >
    <span>
      test-segment-label
      &gt;
    </span>
  </li>
</div>
`;
